### PR TITLE
[ObjC]fix 64 to 32 bit clang conversion warning in src/core/lib/security

### PR DESCRIPTION
**Pull request by HannahShiSFB**
https://github.com/grpc/grpc-ios/issues/83

fix 64 to 32 bit clang conversion warning in src/core/lib/security

@dennycd

### CODE REVIEW

File: src/core/lib/security/credentials/tls/grpc_tls_certificate_provider.cc [hunk 1]
```diff
@@ -101,7 +101,7 @@ gpr_timespec TimeoutSecondsToDeadline(int64_t seconds) {
 
 FileWatcherCertificateProvider::FileWatcherCertificateProvider(
     std::string private_key_path, std::string identity_certificate_path,
-    std::string root_cert_path, unsigned int refresh_interval_sec)
+    std::string root_cert_path, int64_t refresh_interval_sec)
```
Comment (yashykt) [inline src/core/lib/security/credentials/tls/grpc_tls_certificate_provider.cc:104]: uint64_t probably makes more sense here

Comment (HannahShiSFB) [reply src/core/lib/security/credentials/tls/grpc_tls_certificate_provider.cc:104]: We pass it to `gpr_time_xxx` functions where int64_t is consistently used, so maybe it’s better to use int64_t here as well?

**Review by yashykt — APPROVED** [review]

**Review by ZhenLian — APPROVED** [review]

**Comment by dennycd** [comment]
please rebase to take in https://github.com/grpc/grpc/pull/29988 for the build break issue

**Comment by ZhenLian** [comment]
This has some style check failure in presubmit tests...@HannahShiSFB would you mind fixing it? I think this is pretty close!

**Pull request merged**